protocol/rpcconsumer/rpcconsumer.go

```diff
@@ -229,46 +236,37 @@ func (rpcc *RPCConsumer) Start(ctx context.Context, options *rpcConsumerStartOpt
 				// this is locked so we don't race optimizers creation
 				chainMutexes[chainID].Lock()
 				defer chainMutexes[chainID].Unlock()
-				value, exists := optimizers.Load(chainID)
-				if !exists {
-					// doesn't exist for this chain create a new one
-					baseLatency := common.AverageWorldLatency / 2 // we want performance to be half our timeout or better
-					optimizer = provideroptimizer.NewProviderOptimizer(options.strategy, averageBlockTime, baseLatency, options.maxConcurrentProviders)
-					optimizers.Store(chainID, optimizer)
-				} else {
-					var ok bool
-					optimizer, ok = value.(*provideroptimizer.ProviderOptimizer)
-					if !ok {
-						err = utils.LavaFormatError("failed loading optimizer, value is of the wrong type", nil, utils.Attribute{Key: "endpoint", Value: rpcEndpoint.Key()})
-						return err
-					}
+				var loaded bool
+				var err error
+
+				baseLatency := common.AverageWorldLatency / 2 // we want performance to be half our timeout or better
+				tempOptimizer := provideroptimizer.NewProviderOptimizer(options.strategy, averageBlockTime, baseLatency, options.maxConcurrentProviders)
+				optimizer, loaded, err = optimizers.LoadOrStore(chainID, tempOptimizer)
+				if err != nil {
+					optimizer = nil
+					return utils.LavaFormatError("failed loading optimizer", err, utils.LogAttr("endpoint", rpcEndpoint.Key()))
 				}
-				value, exists = consumerConsistencies.Load(chainID)
-				if !exists { // doesn't exist for this chain create a new one
-					consumerConsistency = NewConsumerConsistency(chainID)
-					consumerConsistencies.Store(chainID, consumerConsistency)
-				} else {
-					var ok bool
-					consumerConsistency, ok = value.(*ConsumerConsistency)
-					if !ok {
-						err = utils.LavaFormatError("failed loading consumer consistency, value is of the wrong type", err, utils.Attribute{Key: "endpoint", Value: rpcEndpoint.Key()})
-						return err
-					}
+				if !loaded { // stored, use the new one
```

Comment:
repeating pattern let's use a generics function here

Reply:
After a second look at this, the first too are redundant. Fix is here: c4f31b4f0d1b89fb58ef69a5cd719e72691877f0